pubtools/_content_gateway/push_base.py

```diff
@@ -0,0 +1,463 @@
+        """
+        Responsible for version workflow operations such as Create,
+        Update and delete the product version based on the satisfied condition
+        on content gateway (CGW).
+
+        The product version operation follows the following workflow:
+            - Search for the product record from the product mapping
+            - Search for the product version record from the product_version(pv_mapping) mapping
+            - If `version state` == `present` and record not found in mapping
+
+                - Creates a new version record associated with product on CGW with given data
+                - Add the new record into the mapping records
+
+            - If `version state` == `present` and record found in mapping
+
+                - Update the version record on content gateway with given data
+                - Update the record into the version mapping records
+
+            - If `version state` == `absent` and record found in mapping
+
+                - Delete the version record on the content gateway
+                - Remove the record from the version mapping records
+
+            - If `version state` == `absent` and record not found in mapping
+
+                - Raises CGWError error
+
+        Args:
+            item (list(dict)):
+                List of version record
+
+        Raises:
+            CGWError:
+                When version record is not present and tries to delete the same
+
+        Returns:
+            version_id (int|None):
+                Returns the version ID
+        """
+
+        product_name = item.get('metadata')['productName']
+        product_code = item.get('metadata')['productCode']
+        version_name = item.get('metadata')['versionName']
+
+        product_id = self.product_mapping.get((product_name, product_code))
+        if not product_id:
+            raise CGWError("Product name: %s product code: %s not found" % (product_name, product_code))
+
+        LOG.debug("Fetching the product and version_id of "
+                  "product_name:- %s and version_name:- %s" % (product_name, version_name))
+
+        version_id = self.pv_mapping.get((product_name, product_code, version_name))
+        if item.get('state') == 'present':
+            item.get('metadata')['productId'] = product_id
+            del item.get('metadata')['productName'], item.get('metadata')['productCode']
+            if not version_id:
+                LOG.info("No previous entries found for the given product's version")
+                LOG.info("Creating version entry for the given version_name: %s ", version_name)
+                version_id = self.cgw_client.create_version(product_id, item.get('metadata'))
+                # adding new version record to the version mapping
+                self.pv_mapping[(product_name, product_code, version_name)] = version_id
+                LOG.info("New version created with version_id:- %s" % version_id)
+                return
```

Comment:
We don't want to return the version id here?

Reply:
We only need the ids back in case of delete operations to validate the same in test scripts. For Update and create it is not needed.